SharpDenizenTools/MetaObjects/MetaTypeData.cs

```diff
@@ -0,0 +1,53 @@
+﻿using FreneticUtilities.FreneticExtensions;
+using SharpDenizenTools.MetaHandlers;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace SharpDenizenTools.MetaObjects
+{
+    /// <summary>A generic-less <see cref="MetaTypeData{T}"/> interface.</summary>
+    public interface IMetaTypeData
+    {
+        /// <summary>Getter to create a new instance of this meta type.</summary>
+        Func<MetaObject> Getter { get; }
+
+        /// <summary>All meta docs of this type, by name.</summary>
+        Dictionary<string, MetaObject> Meta {  get; }
+
+        /// <summary>The meta type this data is for.</summary>
+        MetaType Type { get; }
+
+        /// <summary>Registers a new <see cref="MetaTypeData{T}"/> with the provided values.</summary>
+        public static void Register<T>(MetaDocs meta, MetaType type, Dictionary<string, T> storage, Func<T> getter)
+            where T : MetaObject
+        {
+            MetaTypeData<T> newData = new(storage, getter, type);
+            meta.MetaTypesData.Add(type.Name.ToLowerFast(), newData);
+        }
+
+        /// <summary>Registers a new <see cref="MetaTypeData{T}"/> with the provided values.</summary>
+        public static void Register<T>(MetaDocs meta, string type, Dictionary<string, T> storage, Func<T> getter)
+            where T : MetaObject
+        {
+            MetaTypeData<T> newData = new(storage, getter, null);
+            meta.MetaTypesData.Add(type, newData);
+        }
+
+        /// <summary>Creates a new <see cref="MetaObject"/> using <see cref="Getter"/> and sets it's type.</summary>
+        public MetaObject CreateNewMeta()
+        {
+            MetaObject newMeta = Getter();
+            newMeta.Type = Type;
+            return newMeta;
+        }
+    }
+
+    /// <summary>Data for a specific <see cref="MetaType"/>.</summary>
+    public record MetaTypeData<T>(Dictionary<string, T> Meta, Func<T> Getter, MetaType Type) : IMetaTypeData where T : MetaObject
```

Comment:
`Meta` and `Meta` have the same name on the same instance but different fundamental properties.

The one on the interface should probably be `IReadOnlyDictionary` given that it's... a perpetually generated duplicate of the real data. Ow. maybe should have an `IEnumerable<MetaObject> AllObjects` instead as that's version that's needed and less cursed about it